src/sntp.rs

```diff
@@ -163,6 +168,22 @@ impl EspSntp {
         })
     }
 
+    pub fn subscribe<F: FnMut(core::time::Duration) + Send + 'static>(
+        &mut self,
+        callback: F,
+    ) -> EspSntpSubscription {
```

Comment:
The original idea of `EspSntpSubscription` was to avoid the need for having an `unsubscribe` method (dropping `EspSntpSubscription` should unsubscribe). However to model that you can have only one subscription at a time (or zero), you need to lifetime `EspSntpSubscription` as `EspSntpSubscription<'_>` where `'_` is the lifetime of the `&mut self` reference in the `subscribe` method. Unfortunately, this means you cannot call `get_sync_status` while you have an active subscription!

So I suggest you remove the `EspSntpSubscription` struct altogether.

Furthermore, and in retrospective, there is unfortunately a time race issue with the separate `subscribe` method I did not foresee: since `EspSntp::new` immediately starts the SNTP service, you might miss the first time sync, as it might happen before you call the `subscribe` method. Not sure how much of a problem that would be in practice, but if it is a problem, then unfortunately we need to revert to your                                                        original idea of pushing the callback to one of the constructors. That would imply:
* a third constructor - say - `new_with_callback` that takes `F`
* `unsubscribe` probably becomes a private fn and the only way to unsubscribe is to drop the `EspSntp` instance.

Reply:
Thanks for the review and feedback. Yes, will proceed with the third constructor and follow rest of your points.